_includes/common/security.md

```diff
@@ -58,6 +58,428 @@ You can configure the client's ability to perform each of the following operatio
+
+```js
+// PUT http://localhost:1337/schemas/:className
+// Set the X-Parse-Application-Id and X-Parse-Master-Key header
+// body:
+{
+  classLevelPermissions:
+  {
+    "get": {
+      "*": true, // means Public access
+      "s0meUs3r1d": true, // key must be an id of `_User`
+      "role:admin": true, // key must be `role:${role_name}`
+      "requiresAuthentication": true, // any authenticated users
+      "pointerFields": ["onwer", "followers"] // pointer permissions
+    },
+    "find": {...},    // ... create, update, delete, addField
+    "readUserFields": ["followers", "owner"], // grouped pointer permissions
+    "writeUserFields": ["owner"], // grouped pointer permissions
+    "protectedFields": {
+      "*": ["password","email"],
+       "userField:owner":[]
+    }
+  }
+}
+```
+
+### Public access
+
+`*` - Allows anyone despite authentication status to execute operation.
+
+```js
+{
+  classLevelPermissions:
+  {
+    "get": {
+      "*": true, // Public access
+    }
+  }
+}
+```
+
+### Users, Roles
+
+This works exactly as ACL's
+
+```js
+{
+  classLevelPermissions:
+  {
+    "find": {
+      "s0meUs3r1d": true, // key must be an id of `_User`
+      "role:admin": true, // key must be `role:${role_name}`
+    }
+  }
+}
+```
+
+### Requires Authentication permission (requires parse-server  >= 2.3.0)
+
+Starting version 2.3.0, parse-server introduces a new Class Level Permission `requiresAuthentication`.
+This CLP prevents any non authenticated user from performing the action protected by the CLP.
+
+If you want to restrict access to a full class to only authenticated users, you can use the `requiresAuthentication` class level permission. For example, you want to allow your **authenticated users** to `find` and `get` objects from your application and your admin users to have all privileges, you would set the CLP:
+
+```js
+{
+  classLevelPermissions:
+  {
+    "get": {
+      "requiresAuthentication": true,
+      "role:admin": true
+    },
+    "find": {
+      "requiresAuthentication": true,
+      "role:admin": true
+    },
+    "create": { "role:admin": true },
+    "update": { "role:admin": true },
+    "delete": { "role:admin": true },
+  }
+}
+```
+
+Effects:
+
+* Non authenticated users won't be able to do anything.
+* Authenticated users (any user with a valid sessionToken) will be able to read all the objects in that class
+* Users belonging to the admin role, will be able to perform all operations.
+
+⚠️ Note that this is in no way securing your content, if you allow anyone to login to your server, every client will still be able to query this object.
+
+## Pointer Permissions
+
+While permissions discussed before let you explicitly target a user by id or a role, Pointer Permissions let you dynamically enforce permissions without knowing the id or assigning roles in advance. Instead you define one or more field names of this class, and any users pointed by such fields of a particular object are granted the permission.
```

Comment:
```suggestion
While the permissions discussed before let you explicitly target a user by `id` or a role, Pointer Permissions let you dynamically enforce permissions without knowing the `id` or assigning roles in advance. Instead you define one or more field names of a class, and any users pointed to by such fields of a particular object are granted the permission.
```

Reply:
Well there is a field that points (holds pointers to) user(s) => users are pointed by (or may be 'in') these fields. What do you think?  `to` seems a bit misleading to me

Reply:
`to` seems right to me but perhaps I'm misunderstanding

Reply:
Imagine u have a PullRequest class, with one of its columns 'reviewers'.  in 'reviewers' there are ids of various people having rights to review it. Given a `Pointer<_User>` (_your_id_) is listed in `reviewers` column. Is it ok to say you are pointed to this column? Or you are pointed by/in this column?

Reply:
Ok, I've got it. Personally I think 'pointed to by this column' makes more sense.

Reply:
Changes applied in commit 0bb548e